builtin/client/init.lua

```diff
@@ -10,3 +10,9 @@ dofile(commonpath .. "chatcommands.lua")
 dofile(clientpath .. "chatcommands.lua")
 dofile(clientpath .. "death_formspec.lua")
 dofile(clientpath .. "misc.lua")
+assert(loadfile(commonpath .. "item_s.lua"))({}) -- Just for push/read node callbacks
+
+if core.get_csm_restrictions().read_nodedefs then
+	core.get_content_id = nil
+	core.get_name_from_content_id = nil
```

Comment:
* How do content id <-> node name mappings help getting nodedefs?
* This check can be removed without recompiling minetest.
  (You could make a hash digest over the builtin code.)

Reply:
I was wary of allowing CSMs a new capability (reading content IDs.) If this new capability is not an issue, I can simplify the code here.

Reply:
It seems like there isn't a security issue: https://irc.minetest.net/minetest-dev/2022-09-27#i_6019769